assets/js/googlesitekit/data/create-settings-store.js

```diff
@@ -323,6 +338,36 @@ export const createSettingsStore = (
 				Boolean
 			);
 		},
+
+		/**
+		 * Gets the owned settings slugs for this module.
+		 *
+		 * @since n.e.x.t
+		 *
+		 * @param {Object} state Data store's state.
+		 * @return {Array|null} The array of owned settings slugs for this module if they exist. Returns `null` if no owned settings slugs exist.
+		 */
+		getOwnedSettingsSlugs: ( state ) => {
```

Comment:
Why did we add this selector? It shouldn't be necessary for `haveOwnedSettingsChanged` below since `ownedSettingsSlugs` is already in scope.

Reply:
I did ponder that too, but figured it was in keeping with what I have seen elsewhere in the codebase, for example this use of `getWidgetState` in `isWidgetActive` even though the widget state is in scope for `isWidgetActive` to access directly.

The rationale was to ensure a single source of truth for retrieving widget state (I remember this because I had orginally written `isWidgetActive` to directly reference state but was asked to use `getWidgetState` in the review). Arguably the same could apply here...

https://github.com/google/site-kit-wp/blob/4c3dfbdf9393c9fb2caa5789f6aeec15481840a0/assets/js/googlesitekit/widgets/datastore/widgets.js#L281

Reply:
Hmm, I think the point about not necessarily needing a registry selector would apply to `haveOwnedSettingsChanged` too, wouldn't it? If it was written as `haveOwnedSettingsChanged( state ) {` then `state` would be the slice for that particular store and `selectors.getOwnedSettingsSlugs(state)` or `selectors.haveSettingsChanged(state)` should work...

> I suppose the main difference with your example is that getWidgetState wasn't introduced with isWidgetActive – we use getWidgetState in a number of places.

Yep, that's a fair point. To be honest, I'd be just as happy _not_ adding `getOwnedSettingsSlugs` here, I think it's a bit of a stretch to say we need a separate selector for every last piece of state. Just figured I'd mention what I'd seen previously that could justify this selector.

Pardon me while I make myself comfortable on this fence. :)